mavros/src/plugins/setpoint_raw.cpp

```diff
@@ -216,26 +216,47 @@ class SetpointRawPlugin : public plugin::PluginBase,
 
 	void attitude_cb(const mavros_msgs::AttitudeTarget::ConstPtr &req)
 	{
+		double thrust_scaling;
 		Eigen::Quaterniond desired_orientation;
 		Eigen::Vector3d baselink_angular_rate;
-
-		tf::quaternionMsgToEigen(req->orientation, desired_orientation);
+		Eigen::Vector3d body_rate;
+		double thrust;
+
+		// Set Thrust scaling in px4_config.yaml, setpoint_raw block.
+		// ignore thrust is false by default, unless no thrust scalling is set or thrust is zero
+		auto ignore_thrust = req->thrust != 0.0 && !sp_nh.getParam("thrust_scaling", thrust_scaling);
+
+		if (ignore_thrust) {
+			// I believe it's safer without sending zero thrust, but actually ignoring the actuation.
+			ROS_FATAL("Recieved thrust, but ignore_thrust is true: "
+				"the most likely cause of this is a failure to specify the thrust_scaling parameters "
+				"on px4/apm_config.yaml. Actuation will be ignored.");
+			return;
+		} else {
+			if (thrust_scaling == 0.0) {
+				ROS_WARN("thrust_scaling parameter is set to zero.");
```

Comment:
One more thing, i prefer ROS_xxx_NAMED() loggers in plugins.

Reply:
Done. Gave it the plugin name and set the frequency for once every 5 seconds.